src/jacobian.jl

```diff
@@ -149,3 +149,19 @@ function jacobian_autodiff(f, x::AbstractArray, nonlinfun, alg)
             jac_prototype = jac_prototype, chunksize = chunk_size),
         num_of_chunks)
 end
+
+function simple_jacobian(cache, x::Number)
```

Comment:
shouldn't the nonlinear solve be responsible for providing the jacobian? What if the problem doesn't support autodiff?

Reply:
This is already handled by other dispatches. Why does this exist?

Reply:
yes, however in the current dispatch, the jacobian is calculated at `cache.u`, i.e. the dispatch takes in `cache` as an argument and not the `x`. But in when linesearch is performed, we need the jacobian at multiple `x`'s (for multiple step lengths $\alpha_{k}$), so we can't just do `jacobian(f, cache)` as in the current dispatch. 

for example like here (https://github.com/JuliaNLSolvers/LineSearches.jl/blob/ded667a80f47886c77d67e8890f6adb127679ab4/src/strongwolfe.jl#L84)

i can't think of any other way to approach this, any inputs will be helpful?

Reply:
although i have an idea like this roughly
`function simple_jacobian(cache, x)
    cache.tmp_var = cache.u
    cache.u = x
    J = jacobian(cache, f) ## use our current dispatch
    cache.u = cache.tmp_var
end`

Reply:
this way we don't have to deal with sparsity and AD/Finite diff separately

Reply:
Yes, it should be something like that reusing the existing machinery. But the vjp shouldn't build the Jacobian at all: you don't actually need to calculate the Jacobian for most of this.

Reply:
yes you are absolutely right. so we need $J^{T}v = (v^{T}J)^{T}$. We can calculate vjp $v^{T}J$. are there any routines in `SparseDiffTools.jl` or do we need to do this kind of 'in-situ'

Reply:
if in-situ, can you point me to something that i can take help from regarding reverse-mode AD, i am not very well versed with it.

Reply:
Yes, that's precisely what Reverse-Mode AD calculates without calculating the Jacobian. This is why I want @avik-pal to take a look and see how that would mesh with https://github.com/SciML/NonlinearSolve.jl/pull/206